soroban-env-host/src/host.rs

```diff
@@ -1058,6 +1032,57 @@ impl Host {
         self.record_contract_event(ContractEventType::System, topics, data)?;
         Ok(Status::OK.into())
     }
+
+    fn create_contract(&self, args: CreateContractArgs) -> Result<Object, HostError> {
+        let id_preimage = match args.contract_id {
+            ContractId::Asset(asset) => self.id_preimage_from_asset(asset)?,
+            ContractId::SourceAccount(salt) => self.id_preimage_from_source_account(salt)?,
+            ContractId::Ed25519PublicKey(key_with_signature) => {
+                let signature_payload_preimage = self.create_contract_hash_preimage(
+                    args.source.metered_clone(&self.budget_ref())?,
+                    key_with_signature.salt.metered_clone(self.budget_ref())?,
+                )?;
+                let signature_payload = self.metered_hash_xdr(&signature_payload_preimage)?;
+                self.verify_sig_ed25519(
+                    &mut VmCaller::none(),
+                    self.add_host_object(signature_payload.to_vec())?
+                        .to_object(),
+                    self.add_host_object(key_with_signature.key.0.to_vec())?
+                        .to_object(),
+                    self.add_host_object(key_with_signature.signature.0.to_vec())?
+                        .to_object(),
+                )?;
+                self.id_preimage_from_ed25519(key_with_signature.key, key_with_signature.salt)?
+            }
+        };
+        self.create_contract_with_id_preimage(args.source, id_preimage)
+    }
+
+    fn install_contract(&self, args: InstallContractCodeArgs) -> Result<Object, HostError> {
+        let hash_bytes = self.metered_hash_xdr(&args)?;
+        let hash_obj = self.add_host_object(hash_bytes.to_vec())?.to_object();
+        let code_key = LedgerKey::ContractCode(LedgerKeyContractCode {
+            hash: Hash(hash_bytes.metered_clone(self.budget_ref())?),
+        });
+        if !self.0.storage.borrow_mut().has(&code_key)? {
+            self.with_mut_storage(|storage| {
+                let data = LedgerEntryData::ContractCode(ContractCodeEntry {
+                    hash: Hash(hash_bytes),
+                    code: args.code,
+                    ext: ExtensionPoint::V0,
+                });
+                storage.put(
+                    &code_key,
+                    &LedgerEntry {
+                        last_modified_ledger_seq: 0,
```

Comment:
I think setting this to 0 is fine because `LedgerTxn:commit` will set last modified later on, but I think we should add a comment explaining this? We do this in a few places so maybe a helper that takes just a `LedgerEntryData` and returns a `LedgerEntry` is a good idea. If you'd like, I can open an issue for this instead since it's unrelated to this PR.

Reply:
While it's not strictly relevant, most of the use cases for this are introduced/changed in this PR, so adding a helper makes sense.